hastory-server-data/src/Hastory/Server/Data/UserForm.hs

```diff
@@ -0,0 +1,34 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Hastory.Server.Data.UserForm where
+
+import Data.Aeson
+import qualified Data.Text as T
+import Data.Validity
+import GHC.Generics
+
+import Hastory.Server.Data.Password
+import Hastory.Server.Data.Username
+
+data UserForm =
+  UserForm
+    { userFormUserName :: Username
+    , userFormPassword :: Password
```

Comment:
This is fine by me, but I would used `Text` here to avoid `unsafeShowPassword` because it's not actually an unsafe use.
It's the intended use.